neutron_plugin/rbac_policy.py

```diff
@@ -0,0 +1,198 @@
+        )
+
+        return {}
+
+    return {
+        'object_type': type,
+        'object_id': id
+    }
+
+
+def validate_found_resource(input_dict, found_resource):
+    if found_resource:
+        for key in found_resource.keys():
+            if key in input_dict and input_dict.get(key, None):
+                raise NonRecoverableError(
+                    'Multiple definitions of resource for which '
+                    'RBAC policy should be applied. '
+                    'You specified it both using properties / operation '
+                    'inputs and relationship.'
+                )
+
+
+def create_rbac_policy_object_dict(ctx, args):
+    found_resource = find_resource_to_apply_rbac_policy(ctx)
+    validate_found_resource(
+        ctx.node.properties.get(RBAC_POLICY_OPENSTACK_TYPE, {}),
+        found_resource
+    )
+
+    validate_found_resource(
+        args.get(RBAC_POLICY_OPENSTACK_TYPE, {}),
+        found_resource
+    )
+
+    rbac_policy = create_object_dict(
+        ctx,
+        RBAC_POLICY_OPENSTACK_TYPE,
+        args,
+        found_resource
+    )
+
+    return rbac_policy
+
+
+@operation
+@with_neutron_client
+def create(neutron_client, args, **kwargs):
+    if use_external_resource(ctx, neutron_client, RBAC_POLICY_OPENSTACK_TYPE):
+        return
+
+    rbac_policy_raw = create_rbac_policy_object_dict(ctx, args)
+    ctx.logger.info('rbac_policy: {0}'.format(rbac_policy_raw))
+
+    rbac_policy = rbac_policy_raw.copy()
+    rbac_policy.pop('name', None)  # rbac_policy doesn't accept name parameter
+
+    rp = neutron_client.create_rbac_policy({
+        RBAC_POLICY_OPENSTACK_TYPE: rbac_policy
+    })[RBAC_POLICY_OPENSTACK_TYPE]
+    rp['name'] = None
+
+    set_neutron_runtime_properties(ctx, rp, RBAC_POLICY_OPENSTACK_TYPE)
+
+
+@operation
+@with_neutron_client
+def delete(neutron_client, **kwargs):
+    delete_resource_and_runtime_properties(
+        ctx,
+        neutron_client,
+        COMMON_RUNTIME_PROPERTIES_KEYS
+    )
+
+
+@operation
+@with_neutron_client
+def list_rbac_policies(neutron_client, args, **kwargs):
+    rbac_policies = neutron_client.list_rbac_policies(**args)
+    add_list_to_runtime_properties(
+        ctx,
+        RBAC_POLICY_OPENSTACK_TYPE,
+        rbac_policies.get('rbac_policies', [])
+    )
+
+
+@operation
+@with_neutron_client
+def find_and_delete(neutron_client, args, **kwargs):
+    reference_rbac_policy = create_rbac_policy_object_dict(ctx, args)
+    reference_rbac_policy.pop('name', None)
+    rbac_policies_list = neutron_client.list_rbac_policies() \
+        .get('rbac_policies', [])
+
+    for rbac_policy in rbac_policies_list:
+        if all(
+            item in rbac_policy.items()
+            for item
+            in reference_rbac_policy.items()
+        ):
+            id = rbac_policy['id']
+            ctx.logger.info(
+                'Found RBAC policy with ID: {0} - deleting ...'.format(id)
+            )
```

Comment:
are we saying here that we ONLY want to delete the policy if the current policy list matches the original policy list?